### ISSUE TITLE
 Feature: Support for OSC 133 ANSI escape sequence

### ISSUE THREAD

**Issue by a-kenji** [zellij-org/zellij#899]
Related discussion: #892, #642

https://wezfurlong.org/wezterm/shell-integration.html :

> OSC 133 Escape sequence to define Input, Output and Prompt zones

How could this be implemented?

**Comment by leelavg** [comment]
- Hi, glad that I checked issues before raising a new one 😅 
- May I know the progress of this or whom to ask?
- I'm on `foot` terminal and when I add below to my bash prompt `foot` terminal is able to recognise the prompts and support jumping to them
```
prompt_marker() {
    printf '\e]133;A\e\\'
}

set_ps1() {
    # my current prompt function
}

PS1="$(set_ps1)$(prompt_marker)"
```
- Even with locked interface, it seems zellij isn't sending this sequence to the emulator (or zellij is handling it incorrectly 🤔)

**Comment by tobias-haenel** [comment]
There should at least be some action like `scroll-next-prompt` or `scroll-previous-prompt` similar to `tmux` (https://man7.org/linux/man-pages/man1/tmux.1.html). I imagine that `zellij` needs to keep track of the various semantic parts prompt, input/command, output in the scrollback buffer based on OSC 133 sequences that are emitted by applications like `nushell`. Maybe some `tmux` contributors or people who understand their codebase might give more details on a potential implementation.

**Comment by tummetott** [comment]
Is someone working on this? OSC 133 is so usefull, it would be a killer feature in zellij.

**Comment by beojan** [comment]
Another request for this. It's really useful to be able to scroll to the start of a build command's output when you're dealing with C++ template error messages.

**Comment by fculpo** [comment]
I opened #5460 with an initial OSC 133 integration focused on selection: triple-clicking marked output selects the command and its output while excluding the prompt.

It handles scrollback, reflow, and incomplete commands while preserving existing selection behavior elsewhere. Testing and feedback across shells and terminals are welcome.

**Comment by fculpo** [comment]
This can be closed, merged via #5460

**Comment by beojan** [comment]
Scrolling to the previous prompt would still be useful I think.

**Comment by fculpo** [comment]
Indeed, maybe better to open a specific issue now that the base is implemented ?

**Comment by schrieveslaach** [comment]
@fculpo, thanks for providing #5460. I would be also nice to edit output of last command instead of the whole scrollback. Is that something you implemented to or has it to be done, still?

**Comment by fculpo** [comment]
Only did triple click to copy output (that was my current need when using
very verbose commands), but yeah every related features such as what you
describe would be nice.
